llvm/tools/sycl-post-link/sycl-post-link.cpp

```diff
@@ -411,6 +417,21 @@ std::string saveModuleIR(Module &M, int I, StringRef Suff) {
   return OutFilename;
 }
 
+bool isImportedFunction(const Function &F) {
+  // Functions with definitions are not imported
+  if (!F.isDeclaration())
+    return false;
+
+  bool ReturnValue = true;
+  if (auto NameStr = itaniumDemangle(F.getName())) {
```

Comment:
I'm concerned by the fact that we do full demangling here, but I'm not sure if we can completely avoid it, unfortunately.

I would suggest more early exits to make sure that we avoid demangling as much as possible:
- if a function has no `sycl-module-id` attribute, it means that it had no `sycl_device` attribute on it, meaning that any calls to it were generated by a compiler at some later point. To me, it seems like such function cannot be imported either
- if a function has `spir_kernel` calling convention, we say `false`. SYCL kernels cannot be called by other kernels and therefore can't be considered as imported functions. As I wrote this I've realized that SYCL kernels won't be declarations so we are unlikely to hit some scenario, this bullet can be ignored

Reply:
> if a function has no sycl-module-id attribute, it means that it had no sycl_device attribute on it, meaning that any calls to it were generated by a compiler at some later point. To me, it seems like such function cannot be imported either

Can you give an example of such a function?

Reply:
> Can you give an example of such a function?

See [`SPIRITTAnnotationsPass`](https://github.com/intel/llvm/blob/sycl/llvm/lib/Transforms/Instrumentation/SPIRITTAnnotations.cpp) and corresponding [design document](https://github.com/intel/llvm/blob/sycl/sycl/doc/design/ITTAnnotations.md#compiler-wrapper-apis)

Reply:
Updated to only import SYCL External functions.